### PR TITLE
Tlpso 461 - Fix token renewal bug.  Clarify order and visibility of the methods

### PR DESCRIPTION
**Pull request by dlhaines**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,10 +46,10 @@ Once you have this information it is possible to update the properties file
...
 utility is working.
 
-NOTE: This secure will not be kept in a properties file in this project, 
-but held in the properties file of the application that is using the library. 
-For example, if Canvas Course Manager was to use this library then in CCM's 
-properties file you would find these properties there.
+NOTE: This secure information will not be kept in a properties file in this 
+project, but held in a properties file of the application that is using the 
+library. For example, if Canvas Course Manager was to use this library then 
+include the secure properties in a CCM properties file.
```
Comment (pushyamig) [inline README.md:52]: agreed!

File: pom.xml [hunk 2]
```diff
@@ -50,6 +50,12 @@
 			<scope>test</scope>
 		</dependency>
 
+      <dependency>
+        <groupId>org.mockito</groupId>
+        <artifactId>mockito-core</artifactId>
+        <version>1.10.19</version>
+        <scope>test</scope>
+      </dependency>
```
Comment (pushyamig) [inline pom.xml:59]: I am not seeing this library being used anywhere in this pull request? is this needed?

File: src/main/java/edu/umich/ctools/esb/utils/WAPI.java [hunk 3]
```diff
@@ -175,20 +178,48 @@ public WAPIResultWrapper doRequest(String request,HashMap<String,String> headers
...
+	}
+
+	public WAPIResultWrapper getRequest(String request){
+		M_log.info("getRequest() called with request: " + request);
+		return getRequest(request,null);
+	}
+
+	public WAPIResultWrapper getRequest(String request,HashMap<String,String> headers){
+		return getOrPutRequestWithRetry(GET,request,headers);
+	}
```
Comment (pushyamig) [inline src/main/java/edu/umich/ctools/esb/utils/WAPI.java:198]: can you add some comments what is the purpose of the method overloading?

**Review by pushyamig — APPROVED** [review]
Looks Fine

File: src/main/java/edu/umich/ctools/esb/utils/WAPI.java [hunk 4]
```diff
@@ -330,52 +353,51 @@ public WAPIResultWrapper reportError(int status) {
...
+		String incoming_token = headers.get(AUTHORIZATION);
 
-		if(wrappedResult.getStatus()==HTTP_UNAUTHORIZED){
-			wrappedResult = renewToken();
-			if(wrappedResult.getStatus()==HTTP_SUCCESS){
-				wrappedResult = doPutRequest(request,headers);
-			}
+		// check for invalid situation.
+		if (incoming_token == this.token) {
+			M_log.info("updateAuthHeader: token did not change ???: "+incoming_token);
```
Comment (zqian) [inline src/main/java/edu/umich/ctools/esb/utils/WAPI.java:389]: should this be warning/error instead? Anyway, the token seems to be renewed nevertheless.

**Pull request merged**